models/Source.js

```diff
@@ -529,21 +529,24 @@ class Source extends BaseModel {
     const source = await Source.query()
       .findById(id)
       .withGraphFetched(
-        '[reader, replies(notDeleted), tags(notDeleted), attributions]'
+        '[reader, replies(notDeleted), tags(notDeleted), attributions, readActivities(latestFirst)]'
       )
       .modifiers({
         notDeleted (builder) {
           builder.whereNull('deleted')
+        },
+        latestFirst (builder) {
+          builder.orderBy('published', 'desc')
```

Comment:
I'm not much of a knex or pg user so this is well out of my wheelhouse or area of expertise, but I wonder if adding `builder.limit(1)` here would work to fetch only the latest readActivity? Or is `limit` a Does Not Do What It Sounds Like It Does sort of thing?

Reply:
That was my first thought too, but turns out that is a limitation of the .withGraphFetched function... you cannot use a limit or offset on the modifiers. You can only use it on the main request.

Reply:
Ah, of course. That's a pity. I figured you had a reason for not using it but I felt the need to ask anyway 🙂